python-client/pyproject.toml

```diff
@@ -72,7 +73,8 @@ test = [
     "tensorflow>=2.10.0, <2.11; sys_platform != 'darwin' or platform_machine != 'arm64'",
     "tensorflow-text>=2.10.0, <2.11; platform_machine == 'x86_64' or platform_machine == 'amd64'",
     "mlflow>2",
-    "wandb"
+    "wandb",
+    "tensorflow-io-gcs-filesystem==0.27.0",                                                        # Transitive dependency, needed to make it work for windows (downgrade from 0.33 to 0.27)
```

Comment:
should we add `sys_platform` marker in this case? And let other OS users install more recent versions?

Reply:
otherwise we'll be stuck with python <3.11 forever

Reply:
Well, this version is the recommended one according to https://github.com/tensorflow/io
I checked the matrix there, that's why I fixed it.

Honestly, we'll need to make changes to every tf package to handle 3.11 after.

Reply:
could you also tell what exactly breaks on windows with more recent versions of `tensorflow-io-gcs-filesystem` ?

Reply:
Basically, it cannot be installed. It's trying to find a version, but cannot find one matching the needed requirements.
I cannot find the build error anymore, I must have deleted them while cleaning yesterday.

I checked a bit into the lock file, and from I can see, there is not wheel registeres for 0.33 for win os.
![image](https://github.com/Giskard-AI/giskard/assets/4953724/f4a686b3-bbbb-4b88-8734-e2464fe63e31)

I will try to change againt, I will see if it's still broken

Reply:
OK, I reproduced it in here : https://github.com/Giskard-AI/giskard/pull/1389
https://github.com/Giskard-AI/giskard/actions/runs/6120611136/job/16612786445?pr=1389

And I found the root cause : 

https://github.com/tensorflow/io/issues/1789

Basically, since 0.32, there is no win wheel on pipy, I will see this afternoon for the details.
0.31 should work though